components/heap/third_party/dartino/cmpctmalloc.c

```diff
@@ -237,6 +242,15 @@ struct multi_heap_info {
 
 static ssize_t heap_grow(cmpct_heap_t *heap, free_t **bucket);
 
+#ifdef __XTENSA__
```

Comment:
Would it make sense to define this before the GET_THREAD_LOCAL_TAG macro under the same ifdef?